### ISSUE TITLE
Provide rationale / guidance on whether this is preferred over git-scm.com git version

### ISSUE THREAD

**Issue by joshka** [git-for-windows/git#244]
As a software dev (but not a git dev) the git-for-windows site confuses me as to whether I should chose this over the official git-scm.com installer. Aside from the obvious reason, the git version is higher, I can't find a statement on what this brings as advantages over the version available on git-scm.

Currently the project site shows the features of git, but I can't find a great deal of info on why it exists and how it's different from the "official" release.

I'd like to see some sort of messaging on the main site to indicate that this is one of the following:
  1. Pre-release software that will eventually be the "blessed" version on git-scm.com
  2. Just another fork, but unlikely to ever be the "blessed" version
  3. A fork that is of high quality, but runs ahead of the general release curve on git-scm, patches get contributed back and will eventually see equivalent releases on git-scm.
  4. Something else?

I'd also like to see a little comparison between the versions, and a list of any major outstanding issues that would impact daily usage. This would help me assess whether the reward vs risk is weighted one way or another.

I'm aware that this is RC software. That said, I'm also aware that RC software in this realm is often fairly high quality and often useful for production use. If you were a software developer working on Windows software on a day to day basis, would you choose to use this, or stick with the git-scm.com version (1.9.5 at this time)?

**Comment by dakotahawkins** [comment]
Probably the most important thing to me is that this has a 64-bit build available. 

We just started using Git for a giant repo and pack operations would run out of memory unless you lowered the pack memory limit to like 256mb.

**Comment by dscho** [comment]
@joshka all your questions should be addressed by the existence of [the milestone](https://github.com/git-for-windows/git/milestones). You are already aware that this is a release candidate, even...

**Comment by dscho** [comment]
@dakotahawkins you should consider opening a different ticket for this completely unrelated problem.

**Comment by kostix** [comment]
@joshka, your confusion is understandable, but easy to resolve: the software hosted in this repository is *the same* project which is the software offerred via the "official site" (it's not too official, TBH, but I digress).

Basically the project involved like this:

* A long time ago a project to port "vanilla Git" to the Windows platform was started.

    It relied on the MinGW suite for providing some POSIX compatibility, an on the MSYS suite
  for build/tooling infrastructure, and stuff to help compile software packages Git depends on
  (such as a POSIX-compatible shell interpreter, Perl, Tcl/Tk, GNU core/file utils etc).

    Because of the MSYS, the project acquired the name "msysGit".

* The project has always consisted of two logical parts:

    1. The build environment: the Git source code (plus a set of Windows-specific patches)
  MSYS and MinGW, scripts to fetch the depndent packages.
   1. The binary installer.

    The former is used to produce the latter.  The binary installer is what end-users (you)
  are interested in, while the build environment is what is used to develop Git for Windows itself.

* After some time of development and many releases, it was decided to explicitly assign
  different names to these components to reduce the confusion among the users about what they have to download.

    Hence the build environment retained the "msysGit" name, and the binary installer acquired the "Git for Windows" name (also known by the "GfW" acronym).

* After yet more time and several releases, it was decided that the build environment should be reworked to make the development easier, and -- at the same time -- (re-)base the satellite Git toolset onto MSYS2 -- the next iteration of the MSYS project.

    Hence the Git for Windows project (and its build environment) *transitioned* to this one.
  Now the original msysGit (with 1.9.5 being its last release, IIRC) only receives
  critical bugfixes.

TL;DR

* This project is *the same* software, just undergone substantial internal infrastructure changes.
* It's being developed by the same folks.
* All the new and shiny stuff is being implemented in this project.
* This project is what everyone should be using.
* That it has the RC status is OK since the original GfW was always in the "preview" status
  to underline its experimental and volunteer-driven status.  This has not changed.

Hope this helps and I did not make severe factual errors.

----

On a side note, you're sadly abusing the bugtracker: it's created for bugs/issues, not general questions. You should have posted a message to [this list](https://groups.google.com/forum/#!forum/git-for-windows), or [that](https://groups.google.com/forum/#!forum/msysgit) -- for an older iteration of GfW.

**Comment by joshka** [comment]
Awesome answer Konstantine.
Apologies for raising the issue in the wrong place. I wanted to raise it against the website project but issues are turned off for that repo. If I find some time (in project crunch mode right now) I'll submit a PR on the website project with this answer summarized.

--
Joshka

**Comment by dakotahawkins** [comment]
@dscho Perhaps you misunderstood me. The memory issue is an issue with msysgit, because it doesn't have a 64-bit version available. I was attempting to explain that the 64-bit version of available as a part of this project alleviates that issue.

**Comment by dscho** [comment]
@kostix if you find anything missing in https://github.com/git-for-windows/git/wiki#about, would you terribly mind adding it there? Thanks!

**Comment by nalla** [comment]
There is also a PR already for git-scm.com for the date when the RC becomes "RTM"
